[Alerting][Docs] Updating alerting setup docs (#109285)
elastic/kibana: docs/user/alerting/alerting-setup.asciidoc

[[alerting-authorization]]
=== Authorization

Rules are authorized using an <<api-keys, API key>> associated with the last user to edit the rule. This API key captures a snapshot of the user's privileges at the time of edit and is subsequently used to run all background tasks associated with the rule, including condition checks, like {es} queries, and action executions. The following rule actions will re-generate the API key:

* creating a rule
* enabling a disabled rule
* updating a rule

[IMPORTANT]
==============================================

> If a rule requires certain privileges to run, such as index privileges, keep in mind that if a user without those privileges updates or disables/re-enables the rule, the rule will no longer function. Conversely, if a user with greater or administrator privileges modifies the rule, it will begin running with increased privileges.

Reviewer: ```suggestion
If a rule requires certain privileges, such as index privileges, to run, and a user without those privileges updates, disables, or re-enables the rule, the rule will no longer function. Conversely, if a user with greater or administrator privileges modifies the rule, it will begin running with increased privileges.
```
